testutils/src/main/java/com/amplifyframework/testutils/AmplifyTestBase.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils;
+
+import android.content.Context;
+import androidx.test.core.app.ApplicationProvider;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.util.Scanner;
+
+/**
+ * Test base for Amplify instrumentation tests. Any configurable information should
+ * be stored inside "testconfiguration.json", and obtained via static
+ * {@link AmplifyTestBase#getPackageConfigure(String)} method.
```

Comment:
Hm. Is this intended to be _separate_ from the `amplifyconfiguration.json`? Like, you're just using JSON config files to store runtime properties for the tests, but they're not in anyway related to `amplifyconfiguration.json`, right?

That *might* be alright. We should not do this as a base class though. I'm wondering if YAGNI here. Seems like a lot of centralized infrastructure. Also, these `testconfiguration.json` are going to end up having lots of small details about various parts of the test system.

I think it would be better to have tens, hundres, of small `.json` snippet files littered into a plugin project's `src/test/resources/some-bit-of-config-for-a-unit-test.json` (for unit tests) or `src/androidTest/assets/somebit-of-config-for-an-instrumentation-test.json`.

Reply:
Ah yea, I initially had it as `res/raw/credentials.json` inside aws-storage-s3 module, but I just got the idea to do this from our SDK repo (which also uses `testconfiguration.json`). I can make the changes to make them local to each module again.